2.0-PROPOSAL/README.md

```diff
@@ -0,0 +1,907 @@
+GitHub to add short-lived credentials that the `brigade.js` script can use for
+reporting. This effectively changes the schema of certain event payloads from
+what has been documented by GitHub to something Brigade-specific, which is
+unexpected behavior the author of a `brigade.js` script needs to be aware of. It
+is easy to see this is brittle.
+
+Event gateways could take on the burden of reporting status to upstream event
+sources if they were to gain insight into the status and logs of the events they
+have emitted into Brigade.
+
+Refer to:
+
+* [project.go](https://github.com/brigadecore/brigade/blob/8ffeb29d3e8133826851d8c7471915b7c50fe412/pkg/brigade/project.go#L31-L34)
+* [Issue #1039](https://github.com/brigadecore/brigade/issues/1039)
+
+### Job Sidecars
+
+Support for "sidecar" containers in job pods is a feature the maintainers have
+wished to add for some time. To date, this has not been possible. Kubernetes
+supports multiple containers per pod, but does not permit any one container to
+be designated as "primary" and treated differently than the others. Thus,
+there's no way to shut down all containers and conclude a pod's lifecycle once
+one particular container has exited.
+
+A workaround for this could be to monitor pods for completion of a particular
+container and then delete the entire pod. To date, Brigade has been unable to
+take advantage of that workaround since all job state and logs are coupled to
+the continued existence of its corresponding pod. If the decision to utilize
+Kubernetes as a makeshift data store is revisited, so too can the possibility of
+support for job sidecars.
+
+Refer to:
+
+* [Issue #340](https://github.com/brigadecore/brigade/issues/340)
+
+### Retriable Workflows
+
+Support for workflows that can be retried after a failure and resume where they
+left off _without re-executing jobs that have already succeeded--_ has been a
+popular feature request. Under the Brigade 1.x architecture, implementing this
+had not been deemed feasible, however, the author wishes to revisit this feature
+request if new architecture can improve its tenability.
+
+Refer to:
+
+* [Issue #369](https://github.com/brigadecore/brigade/issues/369)
+* [Issue #995](https://github.com/brigadecore/brigade/issues/995)
+
+### Philosophical Pivots
+
+Our final section outlining the motivations driving Brigade 2.0 tackles two
+purely philosophical course corrections.
+
+First, Brigade currently refers to individual executions of user-defined
+workflows as "builds." This is a misnomer which the maintainers collectively
+regret, as it strongly suggests that Brigade is a CI platform, which it is _not_
+even if it _is_ often used to implement CI workflows. Equally regrettable is
+that the term "build" has too often been used synonymously with "event." The
+author proposes to strike "build" from Brigade's lexicon entirely and speak more
+generically, only in terms of events, each of which has a worker that _handles_
+it.
```

Comment:
It's only a half-pivot, and only in terminology. "Build" is used throughtout the code and "build" and "event" are casually used interchangeably throughout the existing docs. We want to just stop saying "build" and only say "event" because that word is loaded with fewer assumptions.